src/request.ts

```diff
@@ -1116,6 +1188,36 @@ class DatastoreRequest {
   }
 }
 
+function isTransaction(request: DatastoreRequest): request is Transaction {
+  return request instanceof Transaction;
+}
+
+export function getTransactionRequest(
```

Comment:
This function is used by `DatastoreRequest` in `request.ts` and `Transaction` in `transaction.ts`. `Transaction` inherits from `DatastoreRequest` so even though this could be moved to a protected method in `DatastoreRequest`, we want to avoid doing so. This is because adding the `protected` reserved word in Typescript doesn't entirely hide the method from the user in the class like `protected` should.